expression/expressions/like.go

```diff
@@ -103,7 +103,12 @@ func (p *PatternLike) Eval(ctx context.Context, args map[interface{}]interface{}
 		}
 		spattern, ok := pattern.(string)
 		if !ok {
```

Comment:
you can use switch type 
switch x := pattern.(type) 
 case string:
case []byte:
default: